contracts/reputation/Coreputation.sol

```diff
@@ -74,6 +74,47 @@ contract Coreputation is MasterCopyNonUpgradable, CoConsensusModule {
         CoConsensusModule.setupCoConsensus(_coConsensus);
     }
 
+    /**
+     * @notice Insert or update validator information.
+     *
+     * @dev Validator status is marked Staked, if validator status is
+     *      Undefined and he has non zero reputation value.
+     *      Validator is not inserted if his reputation value is zero.
```

Comment:
Validator is marked `Deregistered` if provided reputation is zero (whether `Undefined` or `Staked`, `Slashed` should remain slashed)

Reply:
Question: 
What is the use case of new validator who is in `Undefined` state can join with `zero` reputation value 🤔 ?